File: js/modexp.js
function modulo(a, b) {
    let remainder = a % b;
    let trueModulo = trueMod(a, b);
    let correct = remainder === trueModulo;

    let output = "";

    if (a > -1) {
        output += ` ${a} MOD `;
    } else {
        output += `${a} MOD `;
    }

    if (b > -1) {
        output += ` ${b}, OK? ${correct ? "Y" : "N"}.`;
    } else {
        output += `${b}, OK? ${correct ? "Y" : "N"}.`;
    }

    if (remainder > -1) {
        output += ` R= ${remainder} ,`;
    } else {
        output += ` R=${remainder} ,`;
    }

    if (trueModulo > -1) {
        output += ` T= ${trueModulo}`;
    } else {
        output += ` T=${trueModulo}`;
    }

    console.log(output);
}

function trueMod(adivisor,bdividend) {
    return ((adivisor % bdividend) + bdividend) % bdividend;
}

modulo(10, 3);
modulo(-10, 3);
modulo(10, -3);
modulo(-10, -3);

File: cs/modexp.cs
static void Modulo(int divisor, int dividend)
{
    int remainder = divisor % dividend;
    int trueModulo = TrueModulo(divisor,dividend);
    bool correct = remainder == trueModulo;     

    Console.WriteLine($" {divisor} MOD {dividend} = {(correct ? remainder : trueModulo)} (R = {remainder}, T = {trueModulo})");

}

static int TrueModulo(int divisor, int dividend) => 
    ((divisor % dividend) + dividend) % dividend;

Modulo(10,3);
Modulo(-10,3);
Modulo(10,-3);
Modulo(-10,-3);
